registration-processor/core-processor/registration-processor-uin-generator-stage/src/main/java/io/mosip/registration/processor/stages/uingenerator/stage/UinGeneratorStage.java

```diff
@@ -261,49 +269,52 @@ public MessageDTO process(MessageDTO object) {
-						registrationStatusDto.setSubStatusCode(StatusUtil.UIN_GENERATED_SUCCESS.getCode());
-						String uinStatus = isUinAlreadyPresent ? UINConstants.UIN_UNASSIGNED : UINConstants.UIN_ASSIGNED;
-						sendResponseToUinGenerator(registrationId, uinResponseDto.getResponse().getUin(),
-								uinStatus);
-						isTransactionSuccessful = true;
-						registrationStatusDto.setStatusCode(RegistrationStatusCode.PROCESSED.toString());
-						description.setMessage(PlatformSuccessMessages.RPR_UIN_GENERATOR_STAGE_SUCCESS.getMessage());
-						description.setCode(PlatformSuccessMessages.RPR_UIN_GENERATOR_STAGE_SUCCESS.getCode());
-						description.setTransactionStatusCode(RegistrationTransactionStatusCode.PROCESSED.toString());
-						
-					} else {
-						List<ErrorDTO> errors = idResponseDTO != null ? idResponseDTO.getErrors() : null;
-
-						String statusComment = errors != null ? errors.get(0).getMessage()
-								: UINConstants.NULL_IDREPO_RESPONSE;
-						registrationStatusDto.setStatusComment(trimExceptionMessage
-								.trimExceptionMessage(StatusUtil.UIN_GENERATION_FAILED.getMessage() + statusComment));
-						registrationStatusDto.setSubStatusCode(StatusUtil.UIN_GENERATION_FAILED.getCode());
-						object.setInternalError(Boolean.TRUE);
-						registrationStatusDto.setStatusCode(RegistrationStatusCode.REJECTED.toString());
-						registrationStatusDto.setLatestTransactionStatusCode(registrationStatusMapperUtil
-								.getStatusCode(RegistrationExceptionTypeCode.PACKET_UIN_GENERATION_FAILED));
 						sendResponseToUinGenerator(registrationId, uinResponseDto.getResponse().getUin(),
 								UINConstants.UIN_UNASSIGNED);
-						isTransactionSuccessful = false;
-						description.setMessage(PlatformErrorMessages.RPR_UGS_UIN_UPDATE_FAILURE.getMessage());
-						description.setCode(PlatformErrorMessages.RPR_UGS_UIN_UPDATE_FAILURE.getCode());
-						description.setTransactionStatusCode(registrationStatusMapperUtil
-								.getStatusCode(RegistrationExceptionTypeCode.PACKET_UIN_GENERATION_FAILED));
-						String idres = idResponseDTO != null ? idResponseDTO.toString()
-								: UINConstants.NULL_IDREPO_RESPONSE;
-
-						regProcLogger.error(LoggerFileConstant.SESSIONID.toString(),
-								LoggerFileConstant.REGISTRATIONID.toString(), registrationId,
-								statusComment + "  :  " + idres);
-						object.setIsValid(false);
-					}
 
-				} else {
+						if (idRepoCreateIdentityFlag) {
+							sendResponseToUinGenerator(registrationId, uinResponseDto.getResponse().getUin(),
+									UINConstants.UIN_UNASSIGNED);
+							String uin = idRepoService.getUinByRid(registrationId,
+									utility.getGetRegProcessorDemographicIdentity());
+
+							demographicIdentity.put("UIN", uin);
+
+							IdResponseDTO sendIdRepoWithUin = sendIdRepoWithUin(registrationId,
+									registrationStatusDto.getRegistrationType(), demographicIdentity, uin, description);
+
+							if (!sendIdRepoWithUin.getErrors().isEmpty()) {
```

Comment:
In else case of this also, we need to generate the VID and complete the further processing

Reply:
fixed